eras/shelley/impl/src/Cardano/Ledger/Shelley/Rules/Delpl.hs

```diff
@@ -196,14 +197,18 @@ delplTransition ::
   , Environment (EraRule "POOL" era) ~ PoolEnv era
   , Signal (EraRule "POOL" era) ~ PoolCert (EraCrypto era)
   , TxCert era ~ ShelleyTxCert era
+  , Era era
   ) =>
   TransitionRule (ShelleyDELPL era)
 delplTransition = do
   TRC (DelplEnv slot ptr pp acnt, d, c) <- judgmentContext
   case c of
     ShelleyTxCertPool poolCert -> do
+      cEpoch <- liftSTS $ do
+        ei <- asks epochInfoPure
+        epochInfoEpoch ei slot
```

Comment:
The ultimate goal of this PR should be to make this call done only once! That is why description of the ticket says it should be done somewhere in `LEDGERS` rule.

The point is that we do not want to compute the same value over and over again for every transaction in a block, and here even worse for every certificate in a transaction. (Note that `checkSlotNotTooLate` also does the same computation with `epochInfoEpoch`) In other words we need to reduce number of calls to logic in `epochInfoEpoch` to an absolute minimum.

It does mean that some rules (like `DELPL`) will have to accept both: current slot and current epoch. That is a small price to pay in complexity for avoiding the same calculation done a whole bunch of times during validation of every block.